lib/pharos/cluster_manager.rb

```diff
@@ -132,7 +132,20 @@ def apply_phases
       apply_phase(Phases::ConfigureTelemetry, [master_hosts.first], master: master_hosts.first)
     end
 
-    def apply_reset
+    def apply_reset_hosts(hosts, drain: true, delete: true)
+      master_hosts = sorted_master_hosts
+      apply_phase(Phases::GatherFacts, hosts, ssh: true, parallel: true)
+
+      if drain || delete
+        apply_phase(Phases::ConfigureClient, [master_hosts.first], ssh: true, master: master_hosts.first, parallel: false, optional: true)
+      end
+
+      apply_phase(Phases::Drain, hosts, parallel: false) if drain
+      apply_phase(Phases::DeleteHost, hosts, parallel: false, master: master_hosts.first) if delete
```

Comment:
Drain and delete can only happen if `hosts` does not include any masters, right? Just thinking if we really need both options.